### DISCUSSION TITLE
7번째 미팅 때 논의하고 싶은 내용

### DISCUSSION THREAD

**Discussion by Joseph-Cha** [Joseph-Cha/JustFight_Client#31] [Q&A] [answered]
1. Test Runner
  - 테스트 코드 작성 리뷰(#29 )
  - 테스트 코드 작성 방법

**Comment by jongfeel** [comment, marked as answer]
#29 에 실제 test code가 어떻게 작성되어야 하는데 suggestion을 하나 더 만들어 드렸습니다.

우선 실천해야 할 가치로
- 테스트 코드를 짜면서 내가 설계한 클래스 코드가 테스트 가능한지 검토
- 테스트 코드를 만들어 나가면서 설계에 대한 이해도 높이기
입니다.

아직 TDD는 어려울 수 있으니 알고만 계시고요.

테스트 코드를 짜는 중요 원칙은
`내가 만든 로직의 검증` 임을 잊지 않으면 되고요.
부수적인 준비 과정이나 데이터들은 테스트 코드를 작성하면서 해도 됩니다.
예로 button component나 click 이벤트 핸들러 등록 같은 코드 등이죠.

테스트 코드는 사람이 하는 짓(?)에 대한 자동화도 포함되어 있으므로
반복되는 로직의 검증을 테스트코드를 통해 위임하는 것도
테스트 코드를 만들 때 좋은 포인트 중에 하나입니다.

**Reply by Joseph-Cha** [reply]
넵 ㅎㅎㅎ 감사합니다!!
